wagtail/admin/ui/tables/__init__.py

```diff
@@ -167,6 +168,14 @@ def get_cell_context_data(self, instance, parent_context):
         return context
 
 
+class NumberColumn(Column):
+    """A specialised column that displays numbers with locale-aware formatting"""
+
+    def get_value(self, instance):
+        value = super().get_value(instance)
+        return intcomma(value)
```

Comment:
I know I suggested overriding `get_value()`, but looking at the other columns it seems more common to override the displayed value via `get_cell_context_data()`. Perhaps so that you can still access the machine-friendly value (number) by calling `get_value()`.